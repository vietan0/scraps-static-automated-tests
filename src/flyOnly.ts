export type Supe = {
  name: string;
  fly: boolean;
};

export default function flyOnly(source: Supe[]) {
  return source.filter((c) => c.fly === true);
}
