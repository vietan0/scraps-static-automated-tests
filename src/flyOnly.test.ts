import { expect, test } from 'vitest';

import flyOnly, { Supe } from './flyOnly';

const testData: Supe[] = [
  { name: 'Homelander', fly: true },
  { name: 'Queen Maeve', fly: false },
  { name: 'Stormfront', fly: true },
  { name: 'Soldier Boy', fly: false },
  { name: 'A-Train', fly: false },
  { name: 'Ryan', fly: true },
];

test('flyOnly', () => {
  const actual = flyOnly(testData);
  expect(actual).toMatchSnapshot();
});
