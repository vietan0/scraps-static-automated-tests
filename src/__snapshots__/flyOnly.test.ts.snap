// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`flyOnly 1`] = `
[
  {
    "fly": true,
    "name": "Homelander",
  },
  {
    "fly": true,
    "name": "Stormfront",
  },
  {
    "fly": true,
    "name": "Ryan",
  },
]
`;
